v1/recipe/tests/test_update_recipe.py

```diff
@@ -272,3 +274,269 @@ def test_put_recipe(self):
+                        {
+                            "id": 19,
+                            "numerator": 1.0,
+                            "denominator": 1.0,
+                            "measurement": "can",
+                            "title": "light kidney beans"
+                        },
+                        {
+                            "id": 20,
+                            "numerator": 1.0,
+                            "denominator": 1.0,
+                            "measurement": "whole",
+                            "title": "serrano pepper"
+                        },
+                        {
+                            "id": 21,
+                            "numerator": 1.0,
+                            "denominator": 1.0,
+                            "measurement": "whole",
+                            "title": "white onion"
+                        }
+                    ]
+                },
+                {
+                    "id": 5,
+                    "title": "Beef",
+                    "ingredients": [
+                        {
+                            "id": 22,
+                            "numerator": 1.0,
+                            "denominator": 1.0,
+                            "measurement": "pound",
+                            "title": "ground pork"
+                        },
+                        {
+                            "id": 23,
+                            "numerator": 1.0,
+                            "denominator": 1.0,
+                            "measurement": "pound",
+                            "title": "ground sirloin"
+                        },
+                        {
+                            "id": 24,
+                            "numerator": 1.0,
+                            "denominator": 1.0,
+                            "measurement": "dash",
+                            "title": "kosher salt"
+                        }
+                    ]
+                }
+            ],
+            "directions": 'test',
+            "tags": [{'title': 'hi'}, {'title': 'hello'}],
+            "title": "Recipe name",
+            "info": "Recipe info rgo",
+            "source": "google.com",
+            "prep_time": 60,
+            "cook_time": 60,
+            "servings": 8,
+            "rating": 0,
+            "cuisine": {"id": 1},
+            "course": {"id": 2},
+            "photo": "upload/recipe_photos/food2.jpg"
+        }
+
+        # add data with photo url
+        request = self.factory.patch('/api/v1/recipe/recipes/tasty-chili', data=data, format='json')
+        # print(request)
+        request.user = self.staff
+        response = view(request, slug='tasty-chili')
+        self.assertEqual(response.data.get('info'), 'Recipe info rgo')
+        self.assertTrue(response.data.get('photo').endswith("upload/recipe_photos/food.jpg"))
+        # copy file
+        shutil.copy2('v1/fixtures/test/food.jpg', 'site-media/upload/recipe_photos')
```

Comment:
Do not statically link to `site-media`, they are configurable: https://docs.djangoproject.com/en/4.1/topics/files/.